src/Compilers/Core/Portable/CommandLine/CommonCommandLineParser.cs

```diff
@@ -451,11 +457,25 @@ internal static string RemoveTrailingSpacesAndDots(string path)
                     continue;
                 }
 
-                if (string.Equals(arg, shared, StringComparison.OrdinalIgnoreCase))
+                if (arg.StartsWith(shared, StringComparison.OrdinalIgnoreCase))
                 {
+                    var hasValue = arg.Length > shared.Length && (arg[shared.Length] == ':' || arg[shared.Length] == '=');
```

Comment:
How about `var values = string.Split(new[] { ':', '='}, 2`?

And then something like `if (values.Length != 2)`...